.github/workflows/cmake.yml

```diff
@@ -2,38 +2,41 @@ name: CMake
 
 on: [push]
 
-env:
-  # Customize the CMake build type here (Release, Debug, RelWithDebInfo, etc.)
-  BUILD_TYPE: Release
-
 jobs:
-  build:
-    # The CMake configure and build commands are platform agnostic and should work equally
-    # well on Windows or Mac.  You can convert this to a matrix build if you need
-    # cross-platform coverage.
-    # See: https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/managing-complex-workflows#using-a-build-matrix
-    runs-on: ubuntu-latest
+  build_and_release:
+    strategy:
+        fail-fast: false
+        matrix:
+          # os: [ubuntu-latest, ubuntu-18.04]
+          os : [ubuntu-18.04]
+          compiler: [gcc, clang]
+    runs-on: ${{ matrix.os }}
 
     steps:
     - uses: actions/checkout@v2
+      with:
+        submodules: recursive
 
-    - name: Create Build Environment
-      # Some projects don't allow in-source building, so create a separate build directory
-      # We'll use this as our working directory for all subsequent commands
-      run: cmake -E make_directory ${{runner.workspace}}/build
-
-    - name: Configure dependencies
-    - run: sudo apt-get -yq install libpq-dev postgresql-server-dev-13 
+    - name: Install dependencies
+      run: sudo bash scripts/install_ci.sh
+        
       
-    - name: Build
-      working-directory: ${{runner.workspace}}/build
-      shell: bash
-      # Execute the build.  You can specify a specific target with "--target <NAME>"
-      run: make local_build BUILD_TEST_DIR=$GITHUB_WORKSPACE
-
+    - name: Format
+      continue-on-error: true
+      run: ./run-clang-format/run-clang-format.py -r --clang-format-executable clang-format-9 modules sources stress test third_party
+    
     - name: Test
-      working-directory: ${{runner.workspace}}/build
-      shell: bash
-      # Execute tests defined by the CMake configuration.  
-      # See https://cmake.org/cmake/help/latest/manual/ctest.1.html for more detail
-      run: ctest -C $BUILD_TYPE
+      env:
+        CC: ${{ matrix.compiler }}
+        CC_FOR_BUILD: ${{ matrix.compiler }}
+      run: make run_test
+      
+    - name: Prepare for release
+      run: tar -zcf odyssey.linux-amd64.$(git rev-parse --short HEAD).tar.gz sources
+
+    - name: Release
+      uses: fnkr/github-action-ghr@v1
+      if: startsWith(github.ref, 'refs/tags/') && matrix.os == 'ubuntu-18.04' && matrix.compiler == 'gcc' # We need to release only once
+      env:
+        GHR_PATH: odyssey.linux-amd64.$(git rev-parse --short HEAD).tar.gz
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Will work only after filling `secrets` of repository.

See more info [there](https://docs.github.com/en/actions/reference/encrypted-secrets).